### PR TITLE
feat: email notification for Programme Day One

### PR DESCRIPTION
**Pull request by DorisWongNHS**
With registered account:
image

No registered account:
image

TIS21-5823
TIS21-5960
TIS21-6204
TIS21-5959

### CODE REVIEW

File: src/main/java/uk/nhs/tis/trainee/notifications/service/ProgrammeMembershipService.java [hunk 1]
```diff
@@ -243,28 +245,30 @@ public void addNotifications(ProgrammeMembership programmeMembership)
...
           programmeMembership.getTisId());
 
-      JobDataMap jobDataMap = new JobDataMap();
-      jobDataMap.put(TIS_ID_FIELD, programmeMembership.getTisId());
-      jobDataMap.put(PERSON_ID_FIELD, programmeMembership.getPersonId());
-      jobDataMap.put(PROGRAMME_NAME_FIELD, programmeMembership.getProgrammeName());
-      jobDataMap.put(PROGRAMME_NUMBER_FIELD, programmeMembership.getProgrammeNumber());
-      jobDataMap.put(START_DATE_FIELD, programmeMembership.getStartDate());
-      jobDataMap.put(TEMPLATE_OWNER_FIELD, programmeMembership.getManagingDeanery());
       jobDataMap.put(TEMPLATE_NOTIFICATION_TYPE_FIELD, PROGRAMME_CREATED);
```
Comment (Judge40) [inline src/main/java/uk/nhs/tis/trainee/notifications/service/ProgrammeMembershipService.java:271]: Create a new map from the existing `jobDataMap` map of shared values, otherwise in an edge case where a PM is created on day one we'll include extra variables in the day one notification.

File: src/test/java/uk/nhs/tis/trainee/notifications/service/ProgrammeMembershipServiceTest.java [hunk 2]
```diff
@@ -688,6 +688,41 @@ void shouldAddDirectNotificationsWhenNotExcluded() throws SchedulerException {
...
+        dayOneStringCaptor.capture(),
+        dayOneJobDataMapCaptor.capture(),
+        eq(Date.from(START_DATE.atStartOfDay(timezone).toInstant())));
+
+    String dayOneJobId = dayOneStringCaptor.getValue();
+    String dayOneExpectedJobId = PROGRAMME_DAY_ONE + "-" + TIS_ID;
+    assertThat("Unexpected job id.", dayOneJobId, is(dayOneExpectedJobId));
+
+    JobDataMap dayOneJobDataMap = dayOneJobDataMapCaptor.getValue();
+    assertThat("Unexpected tisId.", dayOneJobDataMap.get(TIS_ID_FIELD), is(TIS_ID));
```
Comment (Judge40) [inline src/test/java/uk/nhs/tis/trainee/notifications/service/ProgrammeMembershipServiceTest.java:705]: Check the map size here to ensure we don't include any additional fields that aren't expected.

File: src/main/resources/templates/email/programme-day-one/v1.0.0.html
Comment (Judge40) [inline src/main/resources/templates/email/programme-day-one/v1.0.0.html:1]: There are a few instances of "TIS Self Service" without the hyphen, change them to "TIS Self-Service" for consistency.

**Review by Judge40 — CHANGES_REQUESTED** [review]

**Review by Judge40 — APPROVED** [review]

**Comment by DorisWongNHS** [comment]
With the pipeline is failing with suspected GH issue, we are happy to merge it to Stage?
